src/cartservice/src/Program.cs

```diff
@@ -23,14 +25,23 @@
     Console.WriteLine("REDIS_ADDR environment variable is required.");
     Environment.Exit(1);
 }
-var cartStore = new RedisCartStore(redisAddress);
 
-// Initialize the redis store
-await cartStore.InitializeAsync();
-Console.WriteLine("Initialization completed");
+builder.Logging
+    .AddOpenTelemetry(options => options.AddOtlpExporter())
+    .AddConsole();
+
+builder.Services.AddSingleton<ICartStore>(x=>
+{
+    var store = new RedisCartStore(x.GetRequiredService<ILogger<RedisCartStore>>(), redisAddress);
+    store.Initialize();
+    return store;
+});
 
-builder.Services.AddSingleton<ICartStore>(cartStore);
 builder.Services.AddSingleton<FeatureFlagHelper>();
+builder.Services.AddSingleton(x => new CartService(x.GetRequiredService<ICartStore>(),
+    new RedisCartStore(x.GetRequiredService<ILogger<RedisCartStore>>(), "badhost:1234"),
```

Comment:
Could explain the reason behind adding the `badhost:1234` here?

Reply:
Please treat it as refactor for  src/cartservice/src/services/CartService.cs. It was used there. Now, I have changed only the constructors.